src/mesh/coordinates.cxx

```diff
@@ -28,23 +28,30 @@ Coordinates::Coordinates(Mesh *mesh)
       G3_23(mesh), G1(mesh), G2(mesh), G3(mesh), ShiftTorsion(mesh),
       IntShiftTorsion(mesh), localmesh(mesh), location(CELL_CENTRE) {
 
-  if (mesh->get(dx, "dx")) {
+  // If the source does not include y-boundary guard cells then communications
+  // are needed to fill them. This provides backwards compatibility even though
+  // the communicated guard cell values may not be correct, because the
+  // integrated shear is not necessarily continuous when poloidal angle goes
+  // from 2pi->0
+  bool need_comms = !mesh->sourceHasYGuards();
+
+  if (localmesh->get(dx, "dx", 0., need_comms)) {
     output_warn.write("\tWARNING: differencing quantity 'dx' not found. Set to 1.0\n");
     dx = 1.0;
   }
 
-  if (mesh->periodicX) {
-    mesh->communicate(dx);
+  if (localmesh->periodicX) {
+    localmesh->communicate(dx);
```

Comment:
We're potentially communicating `dx` above, but also doing so here. I appreciate it's for a different reason, but it looks a bit odd